jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -4817,7 +4817,7 @@ public int offsetByCodePoints(int start, int codePointCount) {
 			if (enableCompression &&  (null == compressionFlag || count >= 0)) {
 				int index = start + codePointCount;
 
-				if (index >= len) {
+				if (index > len) {
```

Comment:
I don't think this is correct.  An `index` == `len` is out of range.  

Code in the other branches of this function (see below) special case `codePointCount == 0` and all treat `index == len` as an exception.

Reply:
The following program says otherwise though:

```
class Test
{
        public static void main(String[] args)
        {
                String s = "a";
                System.out.println("s.length() = " + s.length());
                System.out.println("s.offsetByCodePoints(0, 1) = " + s.offsetByCodePoints(0, 1));
        }
}
```

This prints:

```
s.length() = 1
s.offsetByCodePoints(0, 1) = 1
```

Reply:
You're correct.  I misread the code below.   Below, if `index` is already equal to `len`, the processing the code point will be an out of bounds access.